# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/CanonicalBrowsingContext.cpp
# Commit: b326d33f98cb
# Full Hash: b326d33f98cbc1d08d8ca10609ae24e8c852767c
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2020-08-01 09:48:10
# Regressor Bug: 1613431
# File Overlap Count: 1
# Description:
#   Bug 1613431 - Part 2: Ignore synced setters return value. r=nika
#   
#   Depends on D83645
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83646
# ==============================================================================

diff -r 4d29677fb419 -r b326d33f98cb docshell/base/CanonicalBrowsingContext.cpp
--- a/docshell/base/CanonicalBrowsingContext.cpp	Fri Jul 31 13:36:45 2020 +0000
+++ b/docshell/base/CanonicalBrowsingContext.cpp	Fri Jul 31 13:37:00 2020 +0000
@@ -1143,7 +1143,9 @@
   mCurrentLoad = nullptr;
 
   if (!aForProcessSwitch) {
-    SetCurrentLoadIdentifier(Nothing());
+    // Resetting the current load identifier on a discarded context
+    // has no effect when a document load has finished.
+    Unused << SetCurrentLoadIdentifier(Nothing());
   }
 }
 